blog/views.py

```diff
@@ -0,0 +1,62 @@
+from django.shortcuts import render, redirect
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
+from django.views import View, generic
+from django.db.models import Count
+from django import forms
+from django.urls import reverse, reverse_lazy
+from django.shortcuts import get_object_or_404
+from django.contrib.auth.mixins import LoginRequiredMixin
+
+from .models import Post, Commentary
+from .forms import CommentForm
+
+
+class Posts(generic.ListView):
+    model = Post
+    template_name = "blog/index.html"
+    paginate_by = 5
+    queryset = (
+        Post.objects.order_by("-created_time")
+        .prefetch_related("commentaries")
+        .annotate(commentaries_count=Count("commentaries"))
+    )
+    context_object_name = "post_list"
+
+
+class PostDetailView(View):
+    def get(self, request, pk):
+        post = Post.objects.get(pk=pk)

```

Comment:
Consider using `get_object_or_404` instead of `Post.objects.get` to handle cases where the post with the given `pk` does not exist. This will automatically return a 404 response if the object is not found.